app/controllers/patrons_controller.rb

```diff
@@ -5,22 +5,60 @@ def index
     @patrons = Patron.all
   end
 
+  def show
+    @patron = Patron.find(params[:id])
+  end
+
   def new
     @patron = Patron.new
   end
 
   def create
     @patron = Patron.new(patron_params)
-    if @patron.save
-      redirect_to patrons_path, notice: 'Patron added successfully.'
+    
+    respond_to do |format|
+      if @patron.save
+        format.html { redirect_to @patron, notice: 'Patron was successfully created.' }
+        format.json { render :show, status: :created, location: @patron }
+      else
+        format.html { render :new }
+        format.json { render json: @patron.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def edit
+    @patron = Patron.find(params[:id])
+  end
+  
+  def update
+    @patron = Patron.find(params[:id])
+    if @patron.update(patron_params)
+      redirect_to patrons_path, notice: 'Patron updated successfully.'
     else
-      render :new
+      render :edit
     end
   end
 
-  private
+  def destroy
+    @patron = Patron.find(params[:id])
+    @patron.destroy
+    redirect_to patrons_path, notice: 'Patron removed successfully.'
+  end
+  
 
+  private
   def patron_params
-    params.require(:patron).permit(:name, :id_number)
+    params.require(:patron).permit(:name, :id_number, :email, :password, :password_confirmation)
+  end
+  
+  def authorize_user
+    unless current_user.admin? || @patron == current_user
+      redirect_to root_path, alert: 'You are not authorized to perform this action.'
+    end
```

Comment:
Name of this method should be `authorize_admin`(as per the logic), I would also move this method in ApplicationController as mentioned above
Plus, `@patron == current_user` this will never be true